pkg/ottl/ottlfuncs/README.md

```diff
@@ -58,6 +59,19 @@ Available Editors:
 - [set](#set)
 - [truncate_all](#truncate_all)
 
+### append
+
+`append(target, Optional[value], Optional[values])`
```

Comment:
Lets add a warning in this function's docs that `append` can be used to create heterogeneus slices, and that otel attributes don't support that.